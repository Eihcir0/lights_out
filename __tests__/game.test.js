import renderer from 'react-test-renderer';
import React from 'react';
import Game from './../components/game.jsx';

test('Game renders correctly', () => {
  const tree = renderer.create(
    <Game />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('validCell detects cell out of bounds', () => {
  const a = new Game();
  expect(a.validCell([-1,1])).toEqual(false);
});

test('validCell detects cell in bounds', () => {
  const a = new Game();
  expect(a.validCell([1,1])).toEqual(true);
});


test('detects Winner', () => {
  const a = new Game();
  a.state = ({board:
    [
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1]
    ]
  });
  expect(a.checkVictory()).toEqual(true);
});

test('detects non-Winner', () => {
  const a = new Game();
  a.state = ({board:
    [
      [1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1]
    ]
  });
  expect(a.checkVictory()).toEqual(false);
});

test('handles toggle', () => {
  const a = new Game();
  a.state = ({board:
    [
      [1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1],
      [-1,-1,-1,-1,-1]
    ]
  });
  a.toggleLight(0,0);

  expect(a.state).toEqual(
    {board:
      [
        [-1,1,-1,-1,-1],
        [1,-1,-1,-1,-1],
        [-1,-1,-1,-1,-1],
        [-1,-1,-1,-1,-1],
        [-1,-1,-1,-1,-1]
      ]
    }
  );
});
