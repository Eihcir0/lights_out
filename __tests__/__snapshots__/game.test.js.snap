exports[`test Game renders correctly 1`] = `
<div
  id="lightsout" />
`;
